simc.py

```diff
@@ -0,0 +1,175 @@
+"""
+SimulationCraft update manager.
+"""
+import datetime
+from dataclasses import dataclass
+import glob
+import html.parser
+import logging
+import platform
+import os
+import re
+import subprocess
+from typing import List, Optional, Tuple
+from urllib.request import urlopen, urlretrieve
+
+from settings import settings
+try:
+    from settings_local import settings
+except ImportError:
+    pass
+
+from i18n import _
+
+
+_CURRENT_SIMC_VERSION = '910-01'
+_VERSION_RE = re.compile(
+    r'SimulationCraft (?P<version>\d+-\d+) for World of Warcraft '
+    r'(?P<wow_version>\d+\.\d+\.\d+\.\d+) .+'
+    r'hotfix (?P<hotfix_date>\d{4}-\d{2}-\d{2})/(?P<hotfix_version>\d+)(?:, '
+    r'git build (?P<expansion>\S+) (?P<git_commit>[a-f0-9]{1,7}))?')
+_PACKAGE_RE = re.compile(
+    r'(?P<filename>simc-(?P<version>\d+-\d+)-'
+    r'(?P<platform>[^-]+)-(?P<git_commit>[a-f0-9]{1,7})\.(?:dmg|7z))')
+
+
+@dataclass(frozen=True)
+class SimcVersion:
+    version: str
+    wow_version: str
+    hotfix_date: datetime.date
+    hotfix_version: int
+    expansion: Optional[str] = None
+    git_commit: Optional[str] = None
+
+    @staticmethod
+    def parse(input: str) -> 'Optional[SimcVersion]':
+        match = _VERSION_RE.search(input)
+        if not match:
+            return
+        
+        match = match.groupdict()
+        match['hotfix_version'] = int(match['hotfix_version'])
+        match['hotfix_date'] = datetime.date.fromisoformat(match['hotfix_date'])
+        return SimcVersion(**match)
+
+
+class LinkParser(html.parser.HTMLParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.links = []  # type: List[str]
+
+    def handle_starttag(self, tag, attrs):
+        if tag != 'a':
+            return
+        attrs = dict(attrs)
+        href = attrs.get('href')
+        if href and '/' not in href:
+            self.links.append(href)
+
+
+def get_simc_version(simc_path: str) -> Optional[SimcVersion]:
+    """Gets simc version information."""
+    p = subprocess.run([simc_path], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    return SimcVersion.parse(p.stdout.decode())
+
+
+def simc_platform() -> Optional[str]:
+    """SimC platform name for binary builds."""
+    if platform.system() == 'Windows':
+        return {
+            'AMD64': 'win64',
+            'x86': 'win32',
```

Comment:
I guess you could add the new arm64 architecture in here as well. Simc cli is included and it should work just fine. (Unfortunately I do not have have a system to try it out, both simc and  AutoSimc with it)